<a href="https://colab.research.google.com/github/AliAhmed354/Final-Year-Project/blob/main/FYP.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import pandas as pd



In [30]:
dataf = pd.read_csv('/content/drive/MyDrive/Breast_Cancer_Mortality_Survival_Dataset.csv')
dataf.head(5)

Patient ID   Moth of Birth   Age     Sex   Race Marital Status  \
0       A0012             12  68.0  Female  White        Married   
1       A0013             12  50.0  Female  White        Married   
2       A0014             11  58.0  Female  White       Divorced   
3       A0015              3  58.0  Female  White        Married   
4       A0016              1  47.0  Female  White        Married   

   Occupation Code  Adopted Status T Stage  N Stage  ...  \
0             11.0               1       T1      N1  ...   
1             11.0               0       T2      N2  ...   
2              9.0               0       T3      N3  ...   
3             10.0               0       T1      N1  ...   
4             12.0               0       T2      N1  ...   

               differentiate Grade   A Stage Tumor Size  Estrogen Status  \
0      Poorly differentiated     3  Regional        4.0         Positive   
1  Moderately differentiated     2  Regional       35.0         Positive   
2  Moderately differentiated     2  Regional       63.0         Positive   
3      Poorly differentiated     3  Regional       18.0         Positive   
4      Poorly differentiated     3  Regional       41.0         Positive   

  Progesterone Status Regional Node Examined Reginol Node Positive  \
0            Positive                     24                     1   
1            Positive                     14                     5   
2            Positive                     14                     7   
3            Positive                      2                     1   
4            Positive                      3                     1   

  Survival Months  Mortality_Status  
0              60             Alive  
1              62             Alive  
2              75             Alive  
3              84             Alive  
4              50             Alive  

[5 rows x 21 columns]

In [31]:
dataf.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4024 entries, 0 to 4023
Data columns (total 21 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Patient ID              4024 non-null   object 
 1   Moth of Birth           4024 non-null   int64  
 2   Age                     4015 non-null   float64
 3   Sex                     3947 non-null   object 
 4   Race                    4024 non-null   object 
 5   Marital Status          4024 non-null   object 
 6   Occupation Code         4021 non-null   float64
 7   Adopted Status          4024 non-null   int64  
 8   T Stage                 4024 non-null   object 
 9   N Stage                 4024 non-null   object 
 10  6th Stage               4024 non-null   object 
 11  differentiate           4024 non-null   object 
 12  Grade                   4024 non-null   int64  
 13  A Stage                 4024 non-null   object 
 14  Tumor Size              4021 non-null   

- we have missing values
- some variables are not relevent and can be removed
- variables can be converted to numerical data types



In [32]:
dataf.isnull().sum()

Patient ID                 0
Moth of Birth              0
Age                        9
Sex                       77
Race                       0
Marital Status             0
Occupation Code            3
Adopted Status             0
T Stage                    0
N Stage                    0
6th Stage                  0
differentiate              0
Grade                      0
A Stage                    0
Tumor Size                 3
Estrogen Status            0
Progesterone Status        0
Regional Node Examined     1
Reginol Node Positive      0
Survival Months            0
Mortality_Status           4
dtype: int64

In [33]:
dataf.describe()

Moth of Birth          Age  Occupation Code  Adopted Status  \
count    4024.000000  4015.000000      4021.000000     4024.000000   
mean        6.481362    54.094396         8.427257        0.496769   
std         3.475442    11.417118         4.637345        0.500052   
min         1.000000    30.000000         1.000000        0.000000   
25%         3.000000    47.000000         4.000000        0.000000   
50%         6.000000    54.000000         8.000000        0.000000   
75%        10.000000    61.000000        12.000000        1.000000   
max        12.000000   502.000000        16.000000        1.000000   

             Grade   Tumor Size  Survival Months  
count  4024.000000  4021.000000      4024.000000  
mean      2.150596    30.419299        71.472167  
std       0.638234    21.161080        25.361855  
min       1.000000   -75.000000         1.000000  
25%       2.000000    16.000000        56.000000  
50%       2.000000    25.000000        73.000000  
75%       3.000000    38.000000        90.000000  
max       4.000000   140.000000       760.000000

- max age too high
- -75 can't be a tumor size
- 760 months an outlier




In [47]:
df = dataf.drop(['Patient ID ','Moth of Birth', 'Sex', 'Marital Status', 'Occupation Code', 'Adopted Status'], axis=1, inplace = False )
df.head()


Age   Race T Stage  N Stage 6th Stage              differentiate  Grade  \
0  68.0  White       T1      N1       IIA      Poorly differentiated      3   
1  50.0  White       T2      N2      IIIA  Moderately differentiated      2   
2  58.0  White       T3      N3      IIIC  Moderately differentiated      2   
3  58.0  White       T1      N1       IIA      Poorly differentiated      3   
4  47.0  White       T2      N1       IIB      Poorly differentiated      3   

    A Stage  Tumor Size Estrogen Status Progesterone Status  \
0  Regional         4.0        Positive            Positive   
1  Regional        35.0        Positive            Positive   
2  Regional        63.0        Positive            Positive   
3  Regional        18.0        Positive            Positive   
4  Regional        41.0        Positive            Positive   

  Regional Node Examined Reginol Node Positive  Survival Months  \
0                     24                     1               60   
1                     14                     5               62   
2                     14                     7               75   
3                      2                     1               84   
4                      3                     1               50   

  Mortality_Status  
0            Alive  
1            Alive  
2            Alive  
3            Alive  
4            Alive

In [48]:
df = df.dropna()
df.isnull().sum()

Age                       0
Race                      0
T Stage                   0
N Stage                   0
6th Stage                 0
differentiate             0
Grade                     0
A Stage                   0
Tumor Size                0
Estrogen Status           0
Progesterone Status       0
Regional Node Examined    0
Reginol Node Positive     0
Survival Months           0
Mortality_Status          0
dtype: int64

In [49]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 4007 entries, 0 to 4023
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Age                     4007 non-null   float64
 1   Race                    4007 non-null   object 
 2   T Stage                 4007 non-null   object 
 3   N Stage                 4007 non-null   object 
 4   6th Stage               4007 non-null   object 
 5   differentiate           4007 non-null   object 
 6   Grade                   4007 non-null   int64  
 7   A Stage                 4007 non-null   object 
 8   Tumor Size              4007 non-null   float64
 9   Estrogen Status         4007 non-null   object 
 10  Progesterone Status     4007 non-null   object 
 11  Regional Node Examined  4007 non-null   object 
 12  Reginol Node Positive   4007 non-null   object 
 13  Survival Months         4007 non-null   int64  
 14  Mortality_Status        4007 non-null   objec

In [20]:
df['T Stage '].unique()
df['N Stage'].unique()
df['differentiate'].unique()
df['A Stage'].unique()


array(['Regional', 'Distant'], dtype=object)

In [50]:
df['T Stage ']= df['T Stage '].map({'T1':1, 'T2':2, 'T3':3, 'T4':4})


In [51]:
df['N Stage'] = df['N Stage'].map({'N1':1, 'N2':2, 'N3':3})

In [52]:
df['differentiate'] = df['differentiate'].map({'Poorly differentiated':1, 'Moderately differentiated':2,'Well differentiated':3, 'Undifferentiated':4})



In [53]:
df['A Stage'] = df['A Stage'].map({'Regional':1, 'Distant':2})


In [54]:
df['Estrogen Status'] = df['Estrogen Status'].map({'Negative':1, 'Positive':2})

In [55]:
df['Progesterone Status'] = df['Progesterone Status'].map({'Negative':1, 'Positive':2})


In [58]:
df[['Regional Node Examined', 'Reginol Node Positive']] = df[['Regional Node Examined', 'Reginol Node Positive']].apply(pd.to_numeric,errors='coerce')
df.dropna(inplace=True)
df[['Regional Node Examined', 'Reginol Node Positive']] = df[['Regional Node Examined', 'Reginol Node Positive']].apply(pd.to_numeric,errors='coerce')

df.dropna(inplace=True)
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 4006 entries, 0 to 4023
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Age                     4006 non-null   float64
 1   Race                    4006 non-null   object 
 2   T Stage                 4006 non-null   int64  
 3   N Stage                 4006 non-null   int64  
 4   6th Stage               4006 non-null   object 
 5   differentiate           4006 non-null   int64  
 6   Grade                   4006 non-null   int64  
 7   A Stage                 4006 non-null   int64  
 8   Tumor Size              4006 non-null   float64
 9   Estrogen Status         4006 non-null   int64  
 10  Progesterone Status     4006 non-null   int64  
 11  Regional Node Examined  4006 non-null   float64
 12  Reginol Node Positive   4006 non-null   float64
 13  Survival Months         4006 non-null   int64  
 14  Mortality_Status        4006 non-null   objec

In [59]:
df.describe()

Age     T Stage       N Stage  differentiate        Grade  \
count  4006.000000  4006.000000  4006.000000    4006.000000  4006.000000   
mean     54.096355     1.783824     1.438592       1.867698     2.151273   
std      11.421477     0.764505     0.693540       0.642226     0.638023   
min      30.000000     1.000000     1.000000       1.000000     1.000000   
25%      47.000000     1.000000     1.000000       1.000000     2.000000   
50%      54.000000     2.000000     1.000000       2.000000     2.000000   
75%      61.000000     2.000000     2.000000       2.000000     3.000000   
max     502.000000     4.000000     3.000000       4.000000     4.000000   

           A Stage   Tumor Size  Estrogen Status  Progesterone Status  \
count  4006.000000  4006.000000      4006.000000          4006.000000   
mean      1.022716    30.355467         1.932851             1.826261   
std       0.149015    21.013353         0.250312             0.378933   
min       1.000000   -75.000000         1.000000             1.000000   
25%       1.000000    16.000000         2.000000             2.000000   
50%       1.000000    25.000000         2.000000             2.000000   
75%       1.000000    38.000000         2.000000             2.000000   
max       2.000000   140.000000         2.000000             2.000000   

       Regional Node Examined  Reginol Node Positive  Survival Months  
count             4006.000000            4006.000000      4006.000000  
mean                14.633799               4.159511        71.424114  
std                 18.684500               5.113422        25.382125  
min                  1.000000               1.000000         1.000000  
25%                  9.000000               1.000000        56.000000  
50%                 14.000000               2.000000        73.000000  
75%                 19.000000               5.000000        90.000000  
max               1080.000000              46.000000       760.000000